src/CMakeLists.txt

```diff
@@ -207,14 +210,18 @@ DLAF_addSublibrary(
 # Define DLAF's tridiagonal eigensolver library
 DLAF_addSublibrary(
   tridiagonal_eigensolver
-  SOURCES eigensolver/tridiag_solver/mc.cpp
-          eigensolver/tridiag_solver/kernels.cpp
-          $<$<BOOL:${DLAF_WITH_CUDA}>:eigensolver/tridiag_solver/gpu.cpp>
-          # TODO: This should be moved to GPU_SOURCES once HIP is supported for the
-          # tridiagonal solver.
-          $<$<BOOL:${DLAF_WITH_CUDA}>:eigensolver/tridiag_solver/kernels.cu>
+  SOURCES eigensolver/tridiag_solver/mc.cpp eigensolver/tridiag_solver/kernels.cpp
+          $<$<BOOL:${DLAF_WITH_GPU}>:eigensolver/tridiag_solver/gpu.cpp>
+  GPU_SOURCES eigensolver/tridiag_solver/kernels.cu
   LIBRARIES dlaf.factorization dlaf.multiplication dlaf.permutations dlaf.solver dlaf.core
 )
+# Tmp! hip adds --offload-arch compile flag globally for the CXX language in
+# hip-config. We would like therefore to only link to rocthrust when compiling
+# with the HIP language. However COMPILE_LANGUAGE generator expression is not
+# accepted in a target_link_libraries call. The LINK_LANGUAGE generator
+# expression is therefore used but it doesn't add the include directories to the
+# target it's linked to. This is a workaround for that.
```

Comment:
We should open an issue (in DLA-Future or rocthrust...) about this if this is meant to stay here for this PR.